src/Autoloader.php

```diff
@@ -18,8 +18,39 @@ class Autoloader
      */
     public static function load()
     {
+        static $loaded = array();
+
+        $classLoader = self::getClassLoader();
+
         $namespaces = RewriteRules::namespaceRewrite();
-        spl_autoload_register(function ($class) use ($namespaces) {
+        spl_autoload_register(function ($class) use ($namespaces, $classLoader, &$loaded) {
+            if (isset($loaded[$class])) {
+                return;
+            }
+
+            // @todo: we need here reverse mapping:
+            //    from new class names we need to get old class name
+            //    that we can add alias - then typehints can work
+            //    as expected.
+            if (strpos($class, 'Expressive\\') === 0) {
+                if ($file = $classLoader->findFile($class)) {
+                    \Composer\Autoload\includeFile($file);
+                    class_alias($class, 'Zend\\' . $class, false);
+                }
+            } elseif (strpos($class, 'Laminas\\') === 0) {
```

Comment:
Combine the statements, and return early:

```php
if (strpos($class, 'Expressive\\') === 0
    && ($file = $classLoader->findFile($class))
) {
    \Composer\Autoload\includeFile($file);
    class_alias($class, 'Zend\\' . $class, false);
    return;
}

// next condition here
```

I'd also import the `Composer\Autoload\includeFile` function.

Reply:
@weierophinney 
> Combine the statements, and return early

Actually it's not gonna help, because now we have `elseif`, and if we combine ifs then condition from elseif is going to be checked in case strpos($class, 'Expressive\\') is satisfied but the file is not found.

This whole part is going to be changed anyway - as we need to have reverse mapping, then we will do the same as in append preloader - split class name into segments and try to find quicker the legacy name.

> I'd also import the `Composer\Autoload\includeFile` function.

Unfortunately, we cannot. As importing function is available form PHP 5.6. We've decided to support PHP 5.3 here.